rainier-core/src/main/scala/com/stripe/rainier/core/Generator.scala

```diff
@@ -102,12 +102,31 @@ object Generator {
       def get(implicit r: RNG, n: Numeric[Real]): T = fn(r, n)
     }
 
-  def traverse[T](seq: Seq[Generator[T]]): Generator[Seq[T]] =
-    new Generator[Seq[T]] {
-      val requirements: Set[Real] = seq.flatMap(_.requirements).toSet
-      def get(implicit r: RNG, n: Numeric[Real]): Seq[T] =
+  def traverse[T, U](seq: Seq[T])(
+      implicit toGen: ToGenerator[T, U]
+  ): Generator[Seq[U]] =
+    new Generator[Seq[U]] {
+      val requirements: Set[Real] = seq.flatMap(toGen(_).requirements).toSet
```

Comment:
seems like you could do the `toGen` conversion once and stash that rather than doing it both here and every time there's a `get`?